lib/src/protocol/exceptions.dart

```diff
@@ -0,0 +1,61 @@
+// Tbdex exception codes are each thrown in exactly one place.
+// Each code is prefixed with the name of the file in which it is thrown.
```

Comment:
Not every code is tested as of this PR. In subsequent PRs, we can flesh out tests to cover all cases.